### DISCUSSION TITLE
optional fields

### DISCUSSION THREAD

**Discussion by de-Bat** [datalust/seq-tickets#1709] [Help] [answered]
Hi, 
I need to ingest non-structered logs (log4net) to seq. 
All of the logs have the same prefix (time, log level, thread id, message) so i can extract these properties easily.
However, there some properties that appear in certain messages and I'd like to have them as properties too.
currently, I manually convert the files into CLEF format using regular expression and 'find & replace'
Is there a mechanism that can accept 'optional' properties? 
I tried using the extract option but the cli utility got stucked 
I thought about using template messages but I didn't find how to mark an optional field (maybe I can use several template messages?)
Thanks,
Ron

**Comment by nblumhardt** [comment, marked as answer]
Hi!

Currently no, the extraction pattern feature doesn't support this. It might be a job better suited to Fluent Bit - there's an example in https://blog.datalust.co/tailing-a-log-file-or-folder-with-fluent-bit-and-seq/ that might get the ball rolling.

Let me know if this helps!

Best regards,
Nick

**Reply by de-Bat** [reply]
Hi,
Yes this looks like a good solution to the scenario
Thanks!
